persistence-api/src/main/java/io/stargate/db/query/QueryType.java

```diff
@@ -0,0 +1,23 @@
+package io.stargate.db.query;
+
+public enum QueryType {
+  SELECT,
+  INSERT,
+  UPDATE,
+  DELETE,
+  BATCH,
+  OTHER;
+
+  /** Whether this is the kind of a DML query (whether it modifies data). */
+  public boolean isDML() {
+    switch (this) {
```

Comment:
I'm gonna disagree about making this change. The method is called `isDML`, so I think it's better to define it through what a DML *is* (an insert, update, delete or batch), rather than by what it *isn't*. Typically, more query types are likely to get extracted from `OTHER` over time, and with the proposed changed, we'd have to adapt the `isDML` method even if we're not adding any new DML type.

Reply:
a `switch` is only an internal implementation of the `isDML()` method. The caller should not worry about how it is implemented.

Reply:
Who said I was talking about the caller? I'm talking about code readability. I think it's better to implement a method as directly as possible, so in this case, through what a DML is, not indirectly through what it isn't. It's one less indirection the reader's brain has to do, and no matter how trivial that indirection may seem, I prefer it this way.